examples/tv-casting-app/linux/main.cpp

```diff
@@ -679,16 +679,19 @@ LinuxCommissionableDataProvider gCommissionableDataProvider;
 int main(int argc, char * argv[])
 {
 #if defined(ENABLE_CHIP_SHELL)
-    Engine::Root().Init();
     std::thread shellThread([]() { Engine::Root().RunMainLoop(); });
-    Shell::RegisterCastingCommands();
 #endif
 
     CHIP_ERROR err = CHIP_NO_ERROR;
 
     SuccessOrExit(err = chip::Platform::MemoryInit());
     SuccessOrExit(err = chip::DeviceLayer::PlatformMgr().InitChipStack());
 
+#if defined(ENABLE_CHIP_SHELL)
+    Engine::Root().Init();
```

Comment:
Is this OK to do after the thread that calls `Engine::Root().RunMainLoop();` starts above?

Reply:
kept the original order of shell initialization, but moved Platform init methods before this group